src/Relation/Binary/Reasoning/Syntax.agda

```diff
@@ -0,0 +1,366 @@
+  module ≤-syntax where
+    infixr 2 step-≤
+    step-≤ = forward
+    syntax step-≤ x yRz x≤y = x ≤⟨ x≤y ⟩ yRz
+
+
+  -- Strict partial order syntax
+  module <-syntax where
+    infixr 2 step-<
+    step-< = forward
+    syntax step-< x yRz x<y = x <⟨ x<y ⟩ yRz
+
+
+  -- Subset order syntax
+  module ⊆-syntax where
+    infixr 2 step-⊆
+    step-⊆ = forward
+    syntax step-⊆ x yRz x⊆y = x ⊆⟨ x⊆y ⟩ yRz
+
+
+  -- Strict subset order syntax
+  module ⊂-syntax where
+    infixr 2 step-⊂
+    step-⊂ = forward
+    syntax step-⊂ x yRz x⊂y = x ⊂⟨ x⊂y ⟩ yRz
+
+
+  -- Square subset order syntax
+  module ⊑-syntax where
+    infixr 2 step-⊑
+    step-⊑ = forward
+    syntax step-⊑ x yRz x⊑y = x ⊑⟨ x⊑y ⟩ yRz
+
+
+  -- Strict square subset order syntax
+  module ⊏-syntax where
+    infixr 2 step-⊏
+    step-⊏ = forward
+    syntax step-⊏ x yRz x⊏y = x ⊏⟨ x⊏y ⟩ yRz
+
+
+  -- Divisibility syntax
+  module ∣-syntax where
+    infixr 2 step-∣
+    step-∣ = forward
+    syntax step-∣ x yRz x∣y = x ∣⟨ x∣y ⟩ yRz
+
+
+  -- Single-step syntax
+  module ⟶-syntax where
+    infixr 2 step-⟶
+    step-⟶ = forward
+    syntax step-⟶ x yRz x∣y = x ⟶⟨ x∣y ⟩ yRz
+
+
+  -- Multi-step syntax
+  module ⟶*-syntax where
+    infixr 2 step-⟶*
+    step-⟶* = forward
+    syntax step-⟶* x yRz x∣y = x ⟶*⟨ x∣y ⟩ yRz
+
+
+------------------------------------------------------------------------
+-- Syntax for bidirectional relations
+
+  module _
+    {U : REL B A ℓ₄}
+    (sym : Sym U R)
+    where
+
+    backward : ∀ x {y z} → S y z → U y x → T x z
+    backward x yRz x≈y = forward x yRz (sym x≈y)
+
+    -- Setoid equality syntax
+    module ≈-syntax where
+      infixr 2 step-≈ step-≈˘
+      step-≈ = forward
+      step-≈˘ = backward
+      syntax step-≈  x yRz x≈y = x ≈⟨ x≈y ⟩ yRz
+      syntax step-≈˘ x yRz y≈x = x ≈˘⟨ y≈x ⟩ yRz
+
+
+    -- Container equality syntax
+    module ≋-syntax where
+      infixr 2 step-≋ step-≋˘
+      step-≋ = forward
+      step-≋˘ = backward
+      syntax step-≋  x yRz x≋y = x ≋⟨ x≋y ⟩ yRz
+      syntax step-≋˘ x yRz y≋x = x ≋˘⟨ y≋x ⟩ yRz
+
+
+    -- Other equality syntax
+    module ≃-syntax where
+      infixr 2 step-≃ step-≃˘
+      step-≃ = forward
+      step-≃˘ = backward
+      syntax step-≃  x yRz x≃y = x ≃⟨ x≃y ⟩ yRz
+      syntax step-≃˘ x yRz y≃x = x ≃˘⟨ y≃x ⟩ yRz
+
+
+    -- Apartness relation syntax
+    module #-syntax where
+      infixr 2 step-# step-#˘
+      step-# = forward
+      step-#˘ = backward
+      syntax step-#  x yRz x#y = x #⟨ x#y ⟩ yRz
+      syntax step-#˘ x yRz y#x = x #˘⟨ y#x ⟩ yRz
+
+
+    -- Bijection syntax
+    module ⤖-syntax where
+      infixr 2 step-⤖ step-⬻
+      step-⤖ = forward
+      step-⬻ = backward
+      syntax step-⤖ x yRz x⤖y = x ⤖⟨ x⤖y ⟩ yRz
+      syntax step-⬻ x yRz y⤖x = x ⬻⟨ y⤖x ⟩ yRz
+
+
+    -- Inverse syntax
+    module ↔-syntax where
+      infixr 2 step-↔ step-↔-sym
+      step-↔ = forward
+      step-↔-sym = backward
+      syntax step-↔ x yRz x⤖y = x ↔⟨ x⤖y ⟩ yRz
```

Comment:
Perhaps a bit too much cut-and-paste, here and below, in the name of the 3rd argument? If this ever leaks out (say in the names that Agda suggests), it might make things harder for the users to understand.

Reply:
Nice spot, fixed!